File: ShoppingV3/app/src/main/java/dk/itu/shoppingv3/ItemsDB.java
package dk.itu.shoppingv3;

import android.content.Context;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ItemsDB {
    private static ItemsDB sItemsDB;
    private final HashMap<String, String> itemsMap= new HashMap<String, String>();

    private ItemsDB() { fillItemsDB(); }

    public static ItemsDB get() {
        if (sItemsDB == null) sItemsDB = new ItemsDB();
        return sItemsDB;
    }

    public HashMap<String, String> getItemsDB() {return itemsMap; }

    public String listItems() {
        String r= "";
        for (HashMap.Entry <String, String> item: itemsMap.entrySet())
            r= r+"\n Buy "+item.getKey() + " in: "  + item.getValue();
        return r;
    }

    public int size() { return itemsMap.size(); }
    public String getWhere(String what) {return itemsMap.get(what); }

    public void addItem(String what, String where){
        itemsMap.put(what, where);
    }

    public void removeItem(String what){
        if (itemsMap.get(what) != null)  itemsMap.remove(what);
    }

    public void fillItemsDB() {
        itemsMap.put("coffee", "Irma");
        itemsMap.put("carrots", "Netto");
        itemsMap.put("milk", "Netto");
        itemsMap.put("bread", "bakery");
        itemsMap.put("butter", "Irma");
    }

}

File: ShoppingV3Kotlin/app/src/main/java/dk/itu/myshoppingv3kotlin/ItemsDB.kt
package dk.itu.myshoppingv3kotlin

class ItemsDB private constructor(){
    private val itemsMap:HashMap<String,String> = HashMap<String,String>()

    init { fillItemsDB()}

    companion object {
        private var sItemsDB: ItemsDB? = null

        fun get(): ItemsDB {
            return sItemsDB ?: ItemsDB().also { sItemsDB = it }

        }
    }

    fun listItems(): String {
        var r = ""
        for ((key, value) in itemsMap)
            r = "$r\n Buy $key in: $value"
        return r
    }

    fun size(): Int {
        return itemsMap.size
    }
    fun getWhere(what: String): String? {
        return itemsMap[what]
    }

    fun addItem(what: String, where: String) {
        itemsMap[what] = where
    }

    fun removeItem(what: String) {
        if (itemsMap[what] != null) itemsMap.remove(what)
    }

    private fun fillItemsDB() {
        itemsMap.put("coffee", "Irma")
        itemsMap.put("carrots", "Netto")
        itemsMap.put("milk", "Netto")
        itemsMap.put("bread", "bakery")
        itemsMap.put("butter", "Irma")
    }
}